src/Calculator/Views/MainPage.xaml.cs

```diff
@@ -206,58 +206,58 @@ private void OnAppPropertyChanged(object sender, PropertyChangedEventArgs e)
-                {
-                    Model.CalculatorViewModel.HistoryVM.AreHistoryShortcutsEnabled = false;
-                    EnsureCalculator();
-                    if (Model.PreviousMode != ViewMode.Programmer)
-                    {
+                    case ViewMode.Standard:
+                        EnsureCalculator();
+                        Model.CalculatorViewModel.HistoryVM.AreHistoryShortcutsEnabled = true;
                         m_calculator.AnimateCalculator(NavCategory.IsConverterViewMode(previousMode));
-                    }
-                }
-                else if (NavCategory.IsDateCalculatorViewMode(newValue))
-                {
-                    if (Model.CalculatorViewModel != null)
-                    {
-                        Model.CalculatorViewModel.HistoryVM.AreHistoryShortcutsEnabled = false;
-                    }
-                    EnsureDateCalculator();
-                }
-                else if (newValue == ViewMode.Graphing)
-                {
-                    EnsureGraphingCalculator();
-                    KeyboardShortcutManager.DisableShortcuts(true);
-                }
-                else if (NavCategory.IsConverterViewMode(newValue))
-                {
-                    if (Model.CalculatorViewModel != null)
-                    {
+                        Model.CalculatorViewModel.HistoryVM.ReloadHistory(newValue);
+                        break;
+                    case ViewMode.Scientific:
+                        EnsureCalculator();
+                        Model.CalculatorViewModel.HistoryVM.AreHistoryShortcutsEnabled = true;
+                        if (Model.PreviousMode != ViewMode.Scientific)
+                        {
+                            m_calculator.AnimateCalculator(NavCategory.IsConverterViewMode(previousMode));
+                        }
+                        Model.CalculatorViewModel.HistoryVM.ReloadHistory(newValue);
+                        break;
+                    case ViewMode.Programmer:
                         Model.CalculatorViewModel.HistoryVM.AreHistoryShortcutsEnabled = false;
-                    }
-
-                    EnsureConverter();
-                    if (!NavCategory.IsConverterViewMode(previousMode))
-                    {
-                        m_converter.AnimateConverter();
-                    }
+                        EnsureCalculator();
+                        if (Model.PreviousMode != ViewMode.Programmer)
+                        {
+                            m_calculator.AnimateCalculator(NavCategory.IsConverterViewMode(previousMode));
+                        }
+                        break;
+                    case ViewMode.Graphing:
+                        EnsureGraphingCalculator();
+                        KeyboardShortcutManager.DisableShortcuts(true);
+                        break;
+                    default:
+                        if (NavCategory.IsDateCalculatorViewMode(newValue))
```

Comment:
Looks like `IsDateCalculatorViewMode` [just checks for ViewMode.Date](https://github.com/microsoft/calculator/blob/91adfd8e9e14b9719e51f4c5e2e06380b6f429cc/src/CalcViewModel/Common/NavCategory.cpp#L262-L265), and I don't think that is likely to change. In a future change, we could simplify this and put `ViewMode.Date` in a separate `case`.